rct229/rules/section5/section5rule11.py

```diff
@@ -0,0 +1,125 @@
+                    data["climate_zone"], building
+                ),
+            }
+
+        def create_context_list(self, context, data=None):
+            building = context.baseline
+            # List of all baseline above-grade wall surfaces to become the context for AboveGradeWallRule
+            return [
+                UserBaselineProposedVals(None, surface, None)
+                for surface in find_all("$..surfaces[*]", building)
+                if get_opaque_surface_type(surface) == OST.ABOVE_GRADE_WALL
+            ]
+
+        class AboveGradeWallRule(RuleDefinitionBase):
+            def __init__(self):
+                super(Section5Rule11.BuildingRule.AboveGradeWallRule, self).__init__(
+                    rmrs_used=UserBaselineProposedVals(False, True, False),
+                    required_fields={},
+                )
+
+            def get_calc_vals(self, context, data=None):
+                climate_zone: str = data["climate_zone"]
+                above_grade_wall = context.baseline
+                scc: str = data["surface_conditioning_category_dict"][above_grade_wall["id"]]
+                above_grade_wall_u_factor = above_grade_wall["construction"]["u_factor"]
+
+                target_u_factor = None
+                target_u_factor_res = None
+                target_u_factor_nonres = None
+
+                if scc in [
+                    SCC.EXTERIOR_RESIDENTIAL,
+                    SCC.EXTERIOR_NON_RESIDENTIAL,
+                    SCC.SEMI_EXTERIOR,
+                ]:
+                    target_u_factor = table_G34_lookup(climate_zone, scc, OST.ABOVE_GRADE_WALL)[
+                        "u_value"
+                    ]
+                elif scc == SCC.EXTERIOR_MIXED:
+                    target_u_factor_res = table_G34_lookup(
+                        climate_zone, SCC.EXTERIOR_RESIDENTIAL, OST.ABOVE_GRADE_WALL
+                    )["u_value"]
+                    target_u_factor_nonres = table_G34_lookup(
+                        climate_zone, SCC.EXTERIOR_NON_RESIDENTIAL, OST.ABOVE_GRADE_WALL
+                    )["u_value"]
+                    if target_u_factor_res == target_u_factor_nonres:
+                        target_u_factor = target_u_factor_res
+
+                return {
+                    "above_grade_wall_u_factor": above_grade_wall_u_factor,
+                    "target_u_factor": target_u_factor,
+                    "target_u_factor_res": target_u_factor_res,
+                    "target_u_factor_nonres": target_u_factor_nonres,
+                }
+
+            def manaul_check_required(self, context, calc_vals, data=None):
+                target_u_factor_res = calc_vals["target_u_factor_res"]
+                target_u_factor_nonres = calc_vals["target_u_factor_nonres"]
+
+                return (
+                    target_u_factor_res is not None
+                    and target_u_factor_nonres is not None
+                    and target_u_factor_res != target_u_factor_nonres
```

Comment:
Here, it seems we do not need the None check?
We are checking if `target_u_factor_res != target_u_factor_nonres` regardless of whether they are none or values.